test/models/apple/show_test.rb

```diff
@@ -19,37 +19,96 @@
-      apple_show.reload
+  describe "#podcast_feeder_episodes" do
+    it "should return an array of Episode" do
+      assert_equal apple_show.podcast_feeder_episodes.count, 1
+      assert_equal apple_show.podcast_feeder_episodes.first.class, Episode
     end
 
-    it "doesn't raise an error if the attr is nil" do
-      apple_show.instance_variable_set(:@apple_episode_json, nil)
-      apple_show.instance_variable_set(:@podcast_feeder_episodes, nil)
-      apple_show.instance_variable_set(:@episodes, nil)
-      apple_show.reload
-      assert_nil apple_show.instance_variable_get(:@podcast_feeder_episodes)
-      assert_nil apple_show.instance_variable_get(:@apple_episode_json)
-      assert_nil apple_show.instance_variable_get(:@episodes)
+    it "includes deleted episodes" do
+      Episode.where(podcast_id: podcast.id).first.update!(deleted_at: Time.now.utc)
+
+      assert_equal apple_show.podcast_feeder_episodes.count, 1
+      assert apple_show.podcast_feeder_episodes.first.deleted?
     end
 
-    it "doesn't raise an error if the attr is false" do
-      apple_show.instance_variable_set(:@apple_episode_json, false)
-      apple_show.instance_variable_set(:@podcast_feeder_episodes, false)
-      apple_show.instance_variable_set(:@episodes, false)
-      apple_show.reload
-      assert_nil apple_show.instance_variable_get(:@podcast_feeder_episodes)
-      assert_nil apple_show.instance_variable_get(:@apple_episode_json)
-      assert_nil apple_show.instance_variable_get(:@episodes)
+    it "de-duplicates episodes based on the item_guid" do
+      assert_equal Episode.where(podcast_id: podcast.id).length, 1
+
+      episode = Episode.where(podcast_id: podcast.id).first
+      episode.update!(original_guid: "123", deleted_at: Time.now.utc)
+
+      # add another episode with the same guid
+      episode2 = create(:episode, podcast: podcast, item_guid: "123")
+
+      assert_equal apple_show.podcast_feeder_episodes.count, 1
+      assert_equal apple_show.podcast_feeder_episodes.first.id, episode2.id
+    end
+
+    describe "#sort_by_episode_properties" do
+      it "sorts by multiple attributes in descending priority" do
+        now = Time.now.utc
+
+        recs = [
+          OpenStruct.new(deleted_at: now, published_at: now, created_at: now, id: 1),
+          OpenStruct.new(deleted_at: now, published_at: now + 1.second, created_at: now, id: 2),
+          OpenStruct.new(deleted_at: now, published_at: now + 1.second, created_at: now + 1.second, id: 3),
+          OpenStruct.new(deleted_at: nil, published_at: now + 1.second, created_at: now + 1.second, id: 4),
+          OpenStruct.new(deleted_at: nil, published_at: now + 1.second, created_at: now + 2.second, id: 5),
+          OpenStruct.new(deleted_at: nil, published_at: now + 2.second, created_at: now + 2.second, id: 6)
+        ]
+
+        assert_equal [6, 5, 4, 3, 2, 1], apple_show.sort_by_episode_properties(recs).map(&:id)
```

Comment:
Okay, so this is non-deleted before deleted, then published_at asc, created_at asc.